tests/functional/i/inconsistent/inconsistent_returns_noreturn.py

```diff
@@ -84,11 +84,10 @@ def bug_pylint_8747_wrong(self, s: str) -> int:  # [inconsistent-return-statemen
         except ValueError:
             self._does_return_method()
 
-    def bug_pylint_8747_underhanded(self, s: str) -> int:
-        """Every return is not consistent because self._falsely_no_return_method() returns a value.
-
-        Known false negative because we take the typing into account and not inference
-        contrary to what we usually do."""
+    def bug_pylint_8747_incorrect_annotation_trumps_inference(self, s: str) -> int:
+        """Every return is consistent since pylint does not currently use inference
```

Comment:
This is good enough for now. I would have just removed the whole docstring, as it's kind of confusing (why would need to use inference? we just tell from the AST node that it returns something.)

Reply:
My fault for suggesting the word inference in the test name. Do you mind if I push a trivial change to the test name? We're not paying attention to the incorrect annotation, so we shouldn't say that it "trumps" anything.

Reply:
Potentially resolved by https://github.com/pylint-dev/pylint/pull/8750/commits/d9738b74d3846ab616c3b8b93fdb4df72a5ec003 ?